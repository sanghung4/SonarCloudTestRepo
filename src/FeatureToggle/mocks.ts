import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
