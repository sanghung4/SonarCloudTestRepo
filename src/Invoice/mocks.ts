import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
