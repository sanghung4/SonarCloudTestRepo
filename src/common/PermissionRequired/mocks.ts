import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
