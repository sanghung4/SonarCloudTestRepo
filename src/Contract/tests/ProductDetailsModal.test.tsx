import { fireEvent } from '@testing-library/react';
import { noop } from 'lodash-es';

import ContractProductDetailsModal from 'Contract/ProductDetailsModal';
import { GetContractDetailsMock } from 'Contract/tests/mocks';
import { ContractProduct } from 'generated/graphql';
import { render } from 'test-utils/TestWrapper';

describe('Contract Details - Product Detail Modal', () => {
  const itemData = GetContractDetailsMock.result.data.contract
    .contractProducts![0] as ContractProduct;

  it('Snapshot match with data', () => {
    const { baseElement } = render(
      <ContractProductDetailsModal product={itemData} onClose={noop} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('Snapshot match without data', () => {
    const { baseElement } = render(
      <ContractProductDetailsModal onClose={noop} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('Close Button click test', async () => {
    const testFunction = jest.fn();
    const { findByTestId } = render(
      <ContractProductDetailsModal product={itemData} onClose={testFunction} />
    );
    const closeButtonTop = await findByTestId(
      'contract-product-modal-close-button-top'
    );
    const closeButtonBottom = await findByTestId(
      'contract-product-modal-close-button-bottom'
    );
    fireEvent.click(closeButtonTop);
    fireEvent.click(closeButtonBottom);
    expect(testFunction).toBeCalledTimes(2);
  });
});
