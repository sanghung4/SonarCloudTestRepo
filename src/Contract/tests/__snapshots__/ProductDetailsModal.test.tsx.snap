// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contract Details - Product Detail Modal Snapshot match with data 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-15p4qre-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="mui-1"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd MuiDialog-paperFullWidth css-wabxlo-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-tszvf-MuiTypography-root-MuiDialogTitle-root"
          id="mui-1"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-6xq8em-MuiButtonBase-root-MuiIconButton-root"
            data-testid="contract-product-modal-close-button-top"
            tabindex="0"
            type="button"
          >
            <svg>
              delete.svg
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiBox-root css-1ns5f5t"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft css-tdpsb9-MuiTypography-root"
              data-testid="product-details-header"
            >
              Product Details
            </h5>
          </div>
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg MuiBox-root css-klz80g-MuiContainer-root"
            data-testid="contract-product-details-container"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-o1xwpy-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-2ukslu-MuiGrid-root"
              >
                <img
                  alt="Product picture"
                  class="MuiBox-root css-4w7ia0"
                  data-testid="product-picture"
                  src="https://images.tradeservice.com/ProductImages/DIR100108/OOATEY_31911_SML.jpg"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-3vdko4-MuiGrid-root"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1q478d5-MuiTypography-root"
                  data-testid="mfr-name"
                >
                  Test Brand 1
                </div>
                <div
                  class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
                  data-testid="product-name"
                >
                  Test Item 1
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container css-1mi9bjm-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1tsghma-MuiGrid-root"
                  >
                    <div
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1q478d5-MuiTypography-root"
                    >
                      Part#
                       
                      <div
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1fzpubk-MuiTypography-root"
                        data-testid="part-number"
                      >
                        TESTPART1
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1tsghma-MuiGrid-root"
                  >
                    <div
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1q478d5-MuiTypography-root"
                    >
                      MFR#
                       
                      <div
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1fzpubk-MuiTypography-root"
                        data-testid="mfr-number"
                      >
                        TB1-001-TI1
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-j6excx-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-body1 css-2xxltb-MuiTypography-root"
                >
                  Technical Documents
                </h5>
                <div
                  class="MuiGrid-root MuiGrid-container css-13m3rjs-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="test-doc"
                      href="https://google.com"
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        Test Doc
                      </button>
                    </a>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="testing-procedures"
                      href="*"
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        Testing Procedures
                      </button>
                    </a>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="dos-and-donts"
                      href="*"
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        Do's and don'ts
                      </button>
                    </a>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="warranty"
                      href="*"
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        Warranty
                      </button>
                    </a>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="terms-of-service"
                      href="*"
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        Terms of Service
                      </button>
                    </a>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="additional-details"
                      href="*"
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        Additional Details
                      </button>
                    </a>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="null"
                      href=""
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        
                      </button>
                    </a>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="null"
                      href="#"
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        
                      </button>
                    </a>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-3 css-1x1u5qn-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                      data-testid="no-url"
                      href=""
                      target="_blank"
                    >
                      <button
                        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1e09mci-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
                        >
                          <svg>
                            technical-specs.svg
                          </svg>
                        </span>
                        No URL
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-body1 css-2xxltb-MuiTypography-root"
                >
                  Technical Specifications
                </h5>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth MuiBox-root css-10e7mrv-MuiDivider-root"
                />
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
                  >
                    <div
                      class="MuiTableContainer-root css-x7f8rh-MuiTableContainer-root"
                    >
                      <table
                        class="MuiTable-root css-14mhd3s-MuiTable-root"
                      >
                        <tbody
                          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                        >
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              length: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="length"
                            >
                              10'
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              material: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="material"
                            >
                              CPVC
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              pressure_rating: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="pressure-rating"
                            >
                              400 PSI at 73 Deg F
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              temperature_rating: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="temperature-rating"
                            >
                              180 Deg F
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Applicable Standard: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="applicable-standard"
                            >
                              NSF 14/61
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Application: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="application"
                            >
                              Drinking Water System, Hot and Cold Water Distribution
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Cpvc Pipe: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="cpvc-pipe"
                            >
                              1
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              End Connection: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="end-connection"
                            >
                              Plain x Plain
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Length: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="length"
                            >
                              10'
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
                  >
                    <div
                      class="MuiTableContainer-root css-x7f8rh-MuiTableContainer-root"
                    >
                      <table
                        class="MuiTable-root css-14mhd3s-MuiTable-root"
                      >
                        <tbody
                          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                        >
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Material: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="material"
                            >
                              CPVC
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Material Specification: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="material-specification"
                            >
                              ASTM D1784/D2846
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Nominal Pipe Size: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="nominal-pipe-size"
                            >
                              3/4"
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Pipe: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="pipe"
                            >
                              2
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Pressure Rating: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="pressure-rating"
                            >
                              400 PSI at 73 Deg F
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Schedule: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="schedule"
                            >
                              SDR 11
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Temperature Rating: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="temperature-rating"
                            >
                              180 Deg F
                            </td>
                          </tr>
                          <tr
                            class="MuiTableRow-root css-bqkitb-MuiTableRow-root"
                          >
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-zo3ar0-MuiTableCell-root"
                            >
                              Thickness: 
                            </td>
                            <td
                              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mfg996-MuiTableCell-root"
                              data-testid="thickness"
                            >
                              0.08"
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-16g5rpk-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-disableElevation MuiButtonBase-root  css-rsojkw-MuiButtonBase-root-MuiButton-root"
                  data-testid="contract-product-modal-close-button-bottom"
                  tabindex="0"
                  type="button"
                >
                  Close Details
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Contract Details - Product Detail Modal Snapshot match without data 1`] = `
<body
  style=""
>
  <div />
</body>
`;
